tango_ros_common/tango_ros_common/src/main/java/eu/intermodalics/tango_ros_common/ParameterNode.java

```diff
@@ -107,17 +108,21 @@ public void setPreferencesFromParameterServer() {
         SharedPreferences.Editor editor = mSharedPreferences.edit();
         for (String paramName : mParamNames.keySet()) {
             if (mConnectedNode.getParameterTree().has(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName))) {
-                if (mParamNames.get(paramName) == "boolean") {
-                    Boolean booleanValue = mConnectedNode.getParameterTree().getBoolean(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), false);
-                    editor.putBoolean(paramName, booleanValue);
-                }
-                if (mParamNames.get(paramName) == "int_as_string") {
-                    Integer intValue = mConnectedNode.getParameterTree().getInteger(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), 0);
-                    editor.putString(paramName, intValue.toString());
-                }
-                if (mParamNames.get(paramName) == "string") {
-                    String stringValue = mConnectedNode.getParameterTree().getString(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), "");
-                    editor.putString(paramName, stringValue);
+                try {
+                    if (mParamNames.get(paramName) == "boolean") {
+                        Boolean booleanValue = mConnectedNode.getParameterTree().getBoolean(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), false);
+                        editor.putBoolean(paramName, booleanValue);
+                    }
+                    if (mParamNames.get(paramName) == "int_as_string") {
+                        Integer intValue = mConnectedNode.getParameterTree().getInteger(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), 0);
+                        editor.putString(paramName, intValue.toString());
+                    }
+                    if (mParamNames.get(paramName) == "string") {
+                        String stringValue = mConnectedNode.getParameterTree().getString(NodeNamespaceHelper.BuildTangoRosNodeNamespaceName(paramName), "");
+                        editor.putString(paramName, stringValue);
+                    }
+                } catch (ParameterClassCastException e) {
+                    mLog.error("Preference " + paramName + " can not be set from parameter server. " + e.getMessage());
```

Comment:
Another small note (sorry to hijack the comment): I always recommend try to pass the exception as a second parameter to log.error methods, instead of appending the exception message string. That will give more and more precise information to the user debugging the error.